packages/samples/headless-ssr/src/components/hydration-metadata.tsx

```diff
@@ -0,0 +1,33 @@
+import {FunctionComponent} from 'react';
+import {SearchCSRState, SearchSSRState} from '../common/engine';
+
+export interface HydrationMetadataProps {
+  initialState: SearchSSRState;
+  hydrationResult?: SearchCSRState;
+}
+
+export const HydrationMetadata: FunctionComponent<HydrationMetadataProps> = ({
+  initialState,
+  hydrationResult,
+}) => (
+  <>
+    <div>
+      Hydrated:{' '}
+      <span id="hydrated-indicator">{hydrationResult ? 'yes' : 'no'}</span>
+    </div>
+    <span id="hydrated-msg">
+      Rendered engine with{' '}
```

Comment:
terminology Nit: wouldn't it be 'Rendered page' or 'Hydrated engine' (or the verbose version 'Rendered page with hydrated engine')?

Reply:
This message is also rendered when we don't have the hydrated engine yet and only the SSR state, so I prefer not to say "Hydrated engine".

I do like "Rendered page", I'll change it to that.